### PR TITLE
Migrate from CC Free to CC Dedicated

### PR DESCRIPTION
**Pull request by lnhsingh**
Add instructions on how to migrate from a CC Free cluster to a paid CC cluster.

Closes [CC-3834](https://cockroachlabs.atlassian.net/browse/CC-3834).

### CODE REVIEW

**Comment by lnhsingh** [comment]
@mwang1026 / @kannanlakshmi - going to have Rich review this doc to get it moving, but would still appreciate your feedback on it! If you don't have time, please lmk if there's someone else who can review (I wasn't sure if another PM or eng maybe would be appropriate for reviewing instead)

**Review by kannanlakshmi — COMMENTED** [review]
Hi! I am not really the best person to review this as its bulk-IO work. Michael would be best suited here and I see that he is already cc-ed. 

> Reviewed 1 of 4 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10411#-:-MZT0NA0AFG8x95CPLsP:b-7b55wz)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mwang1026 and @rmloveland)

File: _includes/sidebar-data-v21.1.json [hunk 1]
```diff
@@ -2626,6 +2626,12 @@
                   "/${VERSION}/cockroach-userfile-list.html"
                 ]
               },
+              {
```
Comment (rmloveland) [inline _includes/sidebar-data-v21.1.json:2629]: hmm snuck in?

File: cockroachcloud/free-faqs.md [hunk 2]
```diff
@@ -100,3 +100,5 @@ In the meantime, you can run a [`SELECT`](../v20.2/select.html) statement using
 ~~~
 $ cockroach sql -e 'SELECT * FROM test_database.table1' --format=csv --url='postgres://username:password@free-tier...' > users.txt
 ~~~
+
+For an example on how to use this output to migrate to a paid CockroachCloud cluster, see [Migrate from a CockroachCloud Free (beta) to CockroachCloud Cluster](migrate-from-free-to-dedicated.html).
```
Comment (rmloveland) [inline cockroachcloud/free-faqs.md:104]: your call but if you delete the "beta" from this you never have to update it again

File: cockroachcloud/migrate-from-free-to-dedicated.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+6,0,mssaF9m9cd,LXe0YhgLRrwsmd68P,2bElAgrnp8ue,WNXJpBB0ObpVW,WM,223011111,0.0709,30000.00,3001
+7,0,zmssaF,9m9cdLXe0YhgLRrws,md68P2bElA,grnp8ueWNX,OA,264011111,0.1060,30000.00,3001
+8,0,8RsaCXoEz,mssaF9m9cdLXe0Yh,gLRrwsmd68P2bElAgrnp,8ueWNXJpBB0ObpVWo,VW,022311111,0.0173,30000.00,3001
+9,0,fcW8Rs,aCXoEzmssaF9m9,cdLXe0YhgLRrws,md68P2bElAgrnp8ue,JC,230811111,0.0755,30000.00,3001
+10,0,RsaCXoEzm,ssaF9m9cdLXe0YhgLRr,wsmd68P2bE,lAgrnp8ueWNXJpBB0Ob,PV,082911111,0.1779,30000.00,3001
+~~~
+
+## Step 2. Host the files where the CockroachCloud cluster can access them
+
+After you've exported your Cockroach Free (beta) cluster data to your local machine, you now need to upload the files to a storage location where the paid CockroachCloud cluster can access them. **We recommend using [cloud storage](../{{site.versions["stable"]}}/use-cloud-storage-for-bulk-operations.html).**
```
Comment (rmloveland) [inline cockroachcloud/migrate-from-free-to-dedicated.md:70]: Rhetorical question: are (beta) we (beta) worried (beta) that (beta) users (beta) might (beta) forget (beta) this (beta) service (beta) is (beta) still (beta) in (beta) beta?

Comment (rmloveland) [reply cockroachcloud/migrate-from-free-to-dedicated.md:70]: :-)

File: cockroachcloud/migrate-from-free-to-dedicated.md [hunk 4]
```diff
@@ -0,0 +1,193 @@
...
+      schema_name | table_name | type  | owner  | estimated_row_count
+    --------------+------------+-------+--------+----------------------
+      public      | district   | table | lauren |                  10
+      public      | warehouse  | table | lauren |                   1
+    (2 rows)
+    ~~~
+
+## Step 4. Add any foreign key relationships
+
+Once all of the tables you want to migrate have been imported into the CockroachCloud cluster, add the foreign key relationships. To do this, use [`ALTER TABLE ... ADD CONSTRAINT`](../{{site.versions["stable"]}}/add-constraint.html). For example:
```
Comment (rmloveland) [inline cockroachcloud/migrate-from-free-to-dedicated.md:176]: Could link to foreign key docs

File: _includes/sidebar-data-v21.1.json (hunk 1 above)
Comment (rmloveland) [reply _includes/sidebar-data-v21.1.json:2629]: guess this was missing from the TOC?

**Review by rmloveland — APPROVED** [review]
LGTM and I wrote some things to annoy you but you can ignore them :-)

**Review by lnhsingh — COMMENTED** [review]
TFTR!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10411#-:-MZUH8f51ZUISP0q7IIC:b-814g8k)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @mwang1026 and @rmloveland)

___
*[_includes/sidebar-data-v21.1.json, line 2629 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOv-r1-2629:-MZUGOLO1xgpLjXlNJta:b-cf11w3) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/_includes/sidebar-data-v21.1.json#L2629)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

guess this was missing from the TOC?
</blockquote></details>

I noticed it and pulled it in instead of opening a second PR :grimacing: 
___
*[cockroachcloud/free-faqs.md, line 104 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOw-r1-104:-MZUGVae3zHAME8eFgY4:b-cq7iwe) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/cockroachcloud/free-faqs.md#L104)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

your call but if you delete the "beta" from this you never have to update it again
</blockquote></details>

That is true but we are officially referring to it as CockroachCloud Free (beta) (TM). I will just have to CTRL+SHIFT+F and change all of them eventually... what's one more
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 70 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r1-70:-MZUGj6516vrvURFng0c:b2nhswy) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/cockroachcloud/migrate-from-free-to-dedicated.md#L70)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

:-) 
</blockquote></details>

Yes
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 176 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r1-176:-MZUH85P4aIpnjhEdX3T:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/cockroachcloud/migrate-from-free-to-dedicated.md#L176)):*
<details><summary><i>Previously, rmloveland (Rich Loveland) wrote…</i></summary><blockquote>

Could link to foreign key docs
</blockquote></details>

Done.

**Review by mwang1026 — CHANGES_REQUESTED** [review]
Overall looks good. A few nits.

Curious why we went the route of IMPORT vs. CREATE TABLE and then IMPORT INTO? I know it's a few more steps so maybe that's it.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10411#-:-MZUMLvhDBb8tsPlgfP4:bn0yutl)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kannanlakshmi, @lnhsingh, and @rmloveland)

___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 70 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r1-70:-MZULSboBpOeI0jJaDVU:b-zci2n6) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/cockroachcloud/migrate-from-free-to-dedicated.md#L70)):*
<details><summary><i>Previously, lnhsingh (Lauren Hirata Singh) wrote…</i></summary><blockquote>

Yes
</blockquote></details>

Userfile also should work on Cockroach Cloud, but only after I think 20.2.7 or so where we fixed a bug. Soooo.... ya maybe we revisit once everyone's on latest minor version.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 10 at r2](https://reviewable.io/reviews/cockroachdb/docs/10411#-MZUJv821LNrgvEUv9C8:-MZUJv821LNrgvEUv9C9:b-ub0qqf) ([raw file](https://github.com/cockroachdb/docs/blob/7007410d1d9b424eb82fc5cf7d3a6964f6ac8206/cockroachcloud/migrate-from-free-to-dedicated.md#L10)):*
> ```
> like the ability to enable backups or to import data.
> ```

So this is coming after we upgrade to 21.1. Will we change docs then?

Also, I'd suggest something like `upload data with the IMPORT command` vs `import data`. a LOT of github issues where people talk about "importing data" but they're really just running a bunch of bulk inserts.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 49 at r2](https://reviewable.io/reviews/cockroachdb/docs/10411#-MZULGZB0P70F-RRl4GD:-MZULGZB0P70F-RRl4GE:b-xf5be1) ([raw file](https://github.com/cockroachdb/docs/blob/7007410d1d9b424eb82fc5cf7d3a6964f6ac8206/cockroachcloud/migrate-from-free-to-dedicated.md#L49)):*
> ```
> distrcit
> ```

TYPO TYPO

**Review by lnhsingh — COMMENTED** [review]
I went with IMPORT since it's fewer steps. It could be a follow up mini project to add steps on how to do it with CREATE + IMPORT INTO - I'll add an issue

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10411#-:-MZURVgX7Qy1jIFEIkeR:b-thmta3)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @kannanlakshmi, @mwang1026, and @rmloveland)

___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 70 at r1](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r1-70:-MZUQuf15VWv57I5J5R4:b-fkx62t) ([raw file](https://github.com/cockroachdb/docs/blob/aef33545731e0f5ea9a0a88c7e961f284274aa0d/cockroachcloud/migrate-from-free-to-dedicated.md#L70)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

Userfile also should work on Cockroach Cloud, but only after I think 20.2.7 or so where we fixed a bug. Soooo.... ya maybe we revisit once everyone's on latest minor version.
</blockquote></details>

I'll add it now - I think it's a low risk callout since new CC clusters are now running 20.2.8 I think.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 10 at r2](https://reviewable.io/reviews/cockroachdb/docs/10411#-MZUJv821LNrgvEUv9C8:-MZUQWkb0KfD1YCBMPAl:b-xjbpfl) ([raw file](https://github.com/cockroachdb/docs/blob/7007410d1d9b424eb82fc5cf7d3a6964f6ac8206/cockroachcloud/migrate-from-free-to-dedicated.md#L10)):*
> Will we change docs then?

Yep, that's the plan.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 49 at r2](https://reviewable.io/reviews/cockroachdb/docs/10411#-MZULGZB0P70F-RRl4GD:-MZUQgDo-4Bo_L9eJJP2:bl1hl02) ([raw file](https://github.com/cockroachdb/docs/blob/7007410d1d9b424eb82fc5cf7d3a6964f6ac8206/cockroachcloud/migrate-from-free-to-dedicated.md#L49)):*
<details><summary><i>Previously, mwang1026 (Michael Wang) wrote…</i></summary><blockquote>

> ```
> distrcit
> ```

TYPO TYPO
</blockquote></details>

Whoops - Done.

File: cockroachcloud/migrate-from-free-to-dedicated.md [hunk 5]
```diff
@@ -0,0 +1,193 @@
+---
+title: Migrate from a CockroachCloud Free (beta) to CockroachCloud Cluster
+summary: Learn how to migrate data from a CockroachCloud Free (beta) cluster into a paid CockroachCloud cluster.
+toc: true
+---
+
+This page has instructions for migrating data from a CockroachCloud Free (beta) cluster to a paid CockroachCloud cluster CSV using [`IMPORT`](../{{site.versions["stable"]}}/import.html). You may want to migrate to the paid version of CockroachCloud if:
```
Comment (jseldess) [inline cockroachcloud/migrate-from-free-to-dedicated.md:7]: @lnhsingh, `cluster CSV using...` seems off.

File: cockroachcloud/migrate-from-free-to-dedicated.md [hunk 6]
```diff
@@ -0,0 +1,193 @@
...
+In CockroachCloud Free (beta) clusters, all external service integrations are disabled. This means that if you want to export data, you need to use [`cockroach sql --execute`](../{{site.versions["stable"]}}/cockroach-sql.html#general) to query the data you want to export, and then pipe the data to a local file. For example:
+
+{% include copy-clipboard.html %}
+~~~ shell
+$ cockroach sql \
+--url 'postgres://<username>:<password>@free-tier.<region>.cockroachlabs.cloud:26257?sslmode=verify-full&sslrootcert=<path/to/certs_dir>/cc-ca.crt&options=--cluster=<cluster_name>' \
+--execute "SELECT * FROM tpcc.warehouse" --format=csv > /Users/<username>/<path/to/file>/warehouse.csv
+~~~
+
+By running the above example command above, the following data is exported to the local `warehouse.csv` file:
```
Comment (jseldess) [inline cockroachcloud/migrate-from-free-to-dedicated.md:36]: Need to remove one instance of `above`.

File: cockroachcloud/migrate-from-free-to-dedicated.md [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+
+    {% include copy-clipboard.html %}
+    ~~~ sql
+    > CREATE DATABASE tpcc;
+    ~~~
+
+1. Write an [`IMPORT`](../{{site.versions["stable"]}}/import.html) statement that matches the schema of the table data you're importing.
+
+    {{site.data.alerts.callout_success}}
+    You can use the [`SHOW CREATE TABLE`](../{{site.versions["stable"]}}/show-create.html#show-the-create-table-statement-for-a-table) in the CockroachCloud Free (beta) cluster to view the `CREATE` statement for the table you're migrating.
```
Comment (jseldess) [inline cockroachcloud/migrate-from-free-to-dedicated.md:90]: `the SHOW CREATE TABLE in the...` > `the SHOW CREATE TABLE statement in the...`

**Review by lnhsingh — COMMENTED** [review]
TFTRs!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/docs/10411#-:-MZYY0e_1j2-eHclxrY-:bz5mair)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jseldess, @kannanlakshmi, @mwang1026, and @rmloveland)

___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 7 at r3](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r3-7:-MZYXjru5h612NBjE8Di:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f362bf82ecd892be515a7dbc237ee1103c6ad96b/cockroachcloud/migrate-from-free-to-dedicated.md#L7)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

@lnhsingh, `cluster CSV using...` seems off. 
</blockquote></details>

Done.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 36 at r3](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r3-36:-MZYXqBc4TCDW6eo0PlW:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f362bf82ecd892be515a7dbc237ee1103c6ad96b/cockroachcloud/migrate-from-free-to-dedicated.md#L36)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

Need to remove one instance of `above`.
</blockquote></details>

Done.
___
*[cockroachcloud/migrate-from-free-to-dedicated.md, line 90 at r3](https://reviewable.io/reviews/cockroachdb/docs/10411#-MYvrAR7WS7BkmawGaOx-r3-90:-MZYXzmR3QjjPbCezwtX:b-896fix) ([raw file](https://github.com/cockroachdb/docs/blob/f362bf82ecd892be515a7dbc237ee1103c6ad96b/cockroachcloud/migrate-from-free-to-dedicated.md#L90)):*
<details><summary><i>Previously, jseldess (Jesse Seldess) wrote…</i></summary><blockquote>

`the SHOW CREATE TABLE in the...` > `the SHOW CREATE TABLE statement in the...`
</blockquote></details>

Done.

**Pull request merged**